nrf52-hal-common/src/spim.rs

```diff
@@ -56,6 +57,71 @@ impl_spim_ext!(
 /// - The over-read character is hardcoded to `0`.
 pub struct Spim<T>(T);
 
+impl<T> embedded_hal::blocking::spi::Transfer<u8> for Spim<T> where T: SpimExt
+{
+   type Error = Error;
+
+    fn transfer<'w>(&mut self, words: &'w mut [u8]) -> Result<&'w [u8], Error> {
+        let mut offset:usize = 0;
+        while offset < words.len() {
+            let datalen = min(EASY_DMA_SIZE, words.len() - offset);
+            let dataptr = offset + (words.as_ptr() as usize);
+            offset += EASY_DMA_SIZE;
```

Comment:
This line is correct (I think), but it made me pause at first. Maybe using `datalen` would be a bit clearer.

```suggestion
            offset += datalen;
```

Just a suggestion, feel free to ignore.